### ISSUE TITLE
resinOS 2.3.0 not booting Supervisor on rpi 3

### ISSUE THREAD

**Issue by 20k-ultra** [balena-os/balena-supervisor#1626]
If you provision a rpi 3 device with resinOS 2.3.0 the Suprevisor will fail to start with the following error:

```
Event: Device bootstrap failed, retrying {"error":{"message":"`sqlite` does not support inserting default values. Specify values explicitly or use the `useNullAsDefault` config flag. (see docs http://knexjs.org/#Builder-insert).","stack":"TypeError: `sqlite` does not support inserting default values. Specify values explicitly or use the `useNullAsDefault` config flag. (see docs http://knexjs.org/#Builder-insert).\n    at /usr/src/app/dist/app.js:299:328412\n    at n (/usr/src/app/dist/app.js:47:6670)\n    at r (/usr/src/app/dist/app.js:299:215594)\n    at /usr/src/app/dist/app.js:299:328377\n    at n (/usr/src/app/dist/app.js:47:6670)\n    at r (/usr/src/app/dist/app.js:299:215594)\n    at insert (/usr/src/app/dist/app.js:299:328342)\n    at i.toSQL (/usr/src/app/dist/app.js:35:14706)\n    at i.toSQL (/usr/src/app/dist/app.js:96:18779)\n    at /usr/src/app/dist/app.js:299:225999\n    at r (/usr/src/app/dist/app.js:1:6145)\n    at /usr/src/app/dist/app.js:299:173317\n    at r (/usr/src/app/dist/app.js:1:6145)\n    at i._settlePromiseFromHandler (/usr/src/app/dist/app.js:299:125873)\n    at i._settlePromise (/usr/src/app/dist/app.js:299:126673)\n    at i._settlePromise0 (/usr/src/app/dist/app.js:299:127372)"},"delay":30000}
```

**Comment by cywang117** [comment]
I did a bit of sniffing around and gathered context for this issue:

From the JF thread, the user is running OS v2.3.0+rev2 with Supervisor v6.1.3, from 4 years ago. This error message is thrown from the knex package, which has [included this thrown error since 5 years ago](https://github.com/knex/knex/commit/9114bdd8b350a60092b607d99c8e0a930c49f14f#). v6.1.3 Supervisor includes a version of knex which would throw this issue with undefined value insertions. It wasn't until [Supervisor v7.0.0 (2018-03-07)](https://github.com/balena-os/balena-supervisor/commit/b003f48d7ba226c2f91bed9993d9552490496122) that `useNullAsDefault` flag was added to prevent this thrown issue. Thus, upgrading to an OS with any Supervisor v7.0.0 and later would solve this specific problem. From the [OS changelogs](https://github.com/balena-os/meta-balena/blob/master/CHANGELOG.md#v2110), the earliest OS version that would avoid this problem is v2.11.0, but of course upgrading to the latest OS is always recommended when possible. 

As for why this error is thrown, I can only assume that the Supervisor is expecting a value somewhere while provisioning, but not receiving it, thus trying to insert an undefined value into the DB. It would probably be a good idea to just remove 2.3.0 from releases, since we are trying to promote upgrading OS/Supervisor wherever possible.
